### ISSUE TITLE
vcf file import breaks addressbook

### ISSUE THREAD

**Issue by pboguslawski** [nextcloud/contacts#2797]
### Describe the bug

Note: reproduce in test system to avoid data corruption.

1. Create addressbook in Nextcloud/Contacts with 1-2 test contacts.
2. Extract b.vcf file from [b.zip](https://github.com/nextcloud/contacts/files/8989846/b.zip) and import this vcs file to created addressbook.
3. Address book won't display entries any more and one cannot add new contacts to addressbook.

Some chars in b.vcf have broken encoding but nc should handle it and should not break access to whole addressbook.
Thunderbird allows one to import such vcf file and displays both imported contacts without problems.

Tested in nc v24.0.2  and contacts v4.1.1. Similar problem was notices in earlier versions (v23) also.

### Steps to reproduce

As above.

### Expected behavior

Nc should display imported contacts (with broken chars probably similar to Thunderbid) and addressbook should be accessible.

### Actual behavior

Addressbook is empty in UI (database contains imported contacts). Adding new contacts in UI does not work.

### Contact version

v4.1.1

### Operating system

Debian 11

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database

MariaDB

### Operating system

Debian 11

### List of activated Apps

```shell
n/a
```

### Nextcloud Signing status

```shell
n/a
```

### Configuration report

```shell
n/a
```

**Comment by pboguslawski** [comment]
After futher debuging: problem is caused by 0x19 (decimal 25) character present in contact data stored in VCS file. When importing such file to Nextcloud Contacts (directly using web UI or via Thunderbird with CardDAV), this char is stored in SQL DB without changes as 0x19. Then when such contact data is served using CardDAV, XML response with data contains pure 0x19 char without any encoding which breaks XML parsers (web browser, Thinderbird); see Firefox debug console (F12) for example when refreshing addressbook contact list after importing example file from this issue.

Seems...

https://stackoverflow.com/questions/20001448/handling-0x19-in-xslt-1-0

...that 0x19 char cannot be present in valid XML 1.0 file and may be present in XML 1.1 file if escaped correctly (i.e. `&#x0019;`). Not sure if CardDAV spec allows XML 1.1.

Possible solutions:
* don't allow/skip creating contacts in Nextcloud Database with characters that cannot be handled with CardDAV (seems better - user must consciously correct data before importing); may be dangerous if user won't be noticed that some contacts where skipped during import (i.e. when importing to Nextcloud Contacts DB via Thunderbird using CardDAV not web UI),
* use XML 1.1 (not sure if there are other chars that will cause similar problems even in XML 1.1 and not sure if CardDAV can use XML 1.1 and even if so - some clients may be incompatible with it),
* replace invalid characters with U+FFFD before storing it in Nextcloud Database (probably dangerous - data on CardDAV client will be different than in Nextcloud Contacts DB).

**Comment by Kelvino9** [comment]
Got the same bug after importing a vcf file on my phone synchronizing back to my Nextcloud instance.

Is there any solution to get the addressbook back? If not then this bug is quite malicious.

**Comment by aluxnimm** [comment]
Same issue when creating a new contact in the Contacts app with an invalid character in the contact notes. The downloaded ics still contains that character and when syncing via CardDAV the nextcloud server will produce an invalid xml response which breaks any CardDAV client. Since the server sends XML 1.0 it needs to filter out all invalid characters not in the range

`#x9 \| #xA \| #xD \| [#x20-#xD7FF] \| [#xE000-#xFFFD] \| [#x10000-#x10FFFF]`

**Comment by pboguslawski** [comment]
Problem still present in v6.1.3.

**Comment by SebastianKrupinski** [comment]
Hi All,

Can someone please post a vcard with the invalid data? Need something at a reference to figure out the best place to fix this.

Seb

**Comment by pboguslawski** [comment]
`b.vcf` that breaks calendar is attached above, in `b.zip` archive.

**Comment by SebastianKrupinski** [comment]
@pboguslawski thank you! I missed that.

**Comment by SebastianKrupinski** [comment]
@pboguslawski 

I had a look at the file. What is the character is this supposed to be? (the one in orange)

![Image](https://github.com/user-attachments/assets/65e92baf-426f-4844-b9c1-6c4c3eae50a1)

**Comment by pboguslawski** [comment]
> What is the character is this supposed to be?

Garbage that may be generated in any buggy software. Importing such things should not break calendar on import. Consider skipping such corrupted contacts on import and just notifying user about error.

**Comment by SebastianKrupinski** [comment]
> Garbage that may be generated in any buggy software. Importing such things should not break calendar on import. Consider skipping such corrupted contacts on import and just notifying user about error.

I don't think so I think its an actual character, I think the word is 'Księgowa' (accountant in polish).

That means that vcard was encoded in a different character set.

**Comment by pboguslawski** [comment]
> I think the word is 'Księgowa' (accountant in polish).

The problematic char is 0x019 and none of [Polish charsets](https://pl.wikipedia.org/wiki/Kodowanie_polskich_znak%C3%B3w) encode letter "ę" as 0x019. It's corrupted char.

**Comment by SebastianKrupinski** [comment]
> The problematic char is 0x019 and none of [Polish charsets](https://pl.wikipedia.org/wiki/Kodowanie_polskich_znak%C3%B3w) encode letter "ę" as 0x019. It's corrupted char.

I agree

**Comment by michel-thomas** [comment]
Hello,
I also have some import error. vcf exported from Nextcloud and trying to import in Thunderbird.
I don't know how to identify which character is/are invalid.
Thanks for your help
